cli/config.go

```diff
@@ -18,11 +19,24 @@ func configFromCache() (*config, error) {
 		return nil, err
 	}
 	var conf config
+
+	conf.Auth = &token{}
 	if err := json.Unmarshal(rd, &conf); err != nil {
 		return nil, err
 	}
+	if conf.Auth != nil && conf.Auth.(*token).User.Email != "" {
+		return &conf, nil
+	}
+
+	conf.Auth = &apiKey{}
+	if err := json.Unmarshal(rd, &conf); err != nil {
+		return nil, err
+	}
+	if conf.Auth != nil && conf.Auth.(*apiKey).KeyID != "" {
+		return &conf, nil
+	}
```

Comment:
Not the prettiest way to do this. Open to suggestions.

Reply:
[suggestion] One way to make this code more self-explanatory might be to pull out the handling for user configs and API key auth configs, which would also make the code more modular. So you could add a method with a documenting name like:
```
func (cf *config) unmarshallUserConf(confBytes []byte) error {
	cf.Auth = &token{}
	if err := json.Unmarshal(confBytes, &cf); err != nil {
		return err
	}
	if cf.Auth != nil && cf.Auth.(*token).User.Email != "" {
		return nil
	}
	return fmt.Errorf("does not contain user config")
}
```
And an analogous method for api key configs.

Reply:
Nice! That would also be the only suggestion I have; no clever way I can think of to attempt unmarshaling to two different types.